### DISCUSSION TITLE
Opinion about making Copilot a commercial application

### DISCUSSION THREAD

**Discussion by lmtonay** [community/community#19047] [Copilot Conversations]
**Hello, GitHub authority,**

I have been using **GitHub Copilot** for a long time as a technical preview. And it helped me a lot. Really it's an excellent plugin for making our work faster. But recently it is saying to start a free trial and pay after the limited period. This decision seems weird to me as _we the general developers contributed and made Copilot better by our codes_, now it's reversely requiring payment from us. We the developers are contributing millions of lines of code to GitHub. GitHub is also providing various facilities to the developers. But making a tool paid which we are accustomed to using regularly is not a good decision indeed. Rather you could limit its facilities and make a free version also. Hope GitHub will take this opinion positively.

Regards,
Labib Mahfuz Tonay
@lmtonay

**Comment by saengmotmi** [comment]
I'm fully agree with your opinion.

**Comment by opportunityyourway** [comment]
Yes, its unfair that our code is being used to power something that we ourselves have no access to

**Reply by martin-braun** [reply]
It's shameful and disrespectful towards us. Microsoft doesn't deserve to be our host if they won't provide free access for a product that resolves from community work.

**Reply by MulleDK19** [reply]
> Yes, its unfair that our code is being used to power something that we ourselves have no access to

When you started using GitHub you agreed to let them use your code.

**Comment by naegelin** [comment]
I agree with @lmtonay . Sounds  like grounds for an unprecedented class action lawsuit to me.  Github already has paid tiers, why not just include co-pilot as part of a pre-existing paid GitHub subscription  if you MUST charge for it?

**Reply by rauzerfabiano** [reply]
Exactly

**Reply by martin-braun** [reply]
There should be a limited free tier and a beta free tier for beta users with the current functions that beta users got used to. A good move would be to improve the product and charge for the improvement.

**Reply by MulleDK19** [reply]
> I agree with @lmtonay . Sounds like grounds for an unprecedented class action lawsuit to me. Github already has paid tiers, why not just include co-pilot as part of a pre-existing paid GitHub subscription if you MUST charge for it?

How is it grounds for a lawsuit, at all? When you start using GitHub, you agree to the terms of service, which states you grant GitHub the rights to use your code.

**Comment by rauzerfabiano** [comment]
I'm fully agree with your opinion too.

**Comment by infozhcom** [comment]
I'm fully agree with your opinion too

**Comment by antoinekm** [comment]
Speaking of this:

> _I said this last year when Copilot was announced. I think it’s gross that a multi-billion dollar company took all the source code we gave them for free, analyzed it, built a tool, and charges $100 a year for it. Anyone who made a commit to GitHub should get this for free._

Source: https://twitter.com/bphogan/status/1539298492887203840

**Comment by CiberNin** [comment]
Me am thinking u is not much big brain.
Thought big think service cost money & many big dev cost would be free?
Me think u also be think u aught work for free cause me civilization contributed lots to u being able to work at all.

**Reply by shentaibai** [reply]
I think there's something wrong with your thinking
Even though the development has costs, AI is trained only after a large number of users submit code. Both sides have made contributions. Why can't it provide convenience for those who contribute

**Comment by SilenceMerely** [comment]
I'm fully agree with your opinion too

**Comment by martin-braun** [comment]
I fully agree 1000%!!!

It's an absolute low move to allow free access to anyone in the beta and now ask for money to continue to use it. The "worst, but acceptable" move would be to ask **new** users for money, but keep the current features for free for beta users, forever.

We beta users got used to it and now they rip it away from us. Instead they should improve the product and charge for the upgrade.

To be honest, it should be free for beta users as well as users who decide to contribute with their own repositories by allowing Copilot scans.

**Comment by ShinChven** [comment]
Let put it into XGP as well. (Just a joke)

**Comment by martin-braun** [comment]
![image](https://user-images.githubusercontent.com/24259245/184644892-7525a986-552d-46b0-9efb-e076219dc484.png)
